File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-1.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        totalSum = nums[0]
        minNumElements = float('inf')
        left = 0
        right = 0

        while left <= right < len(nums):
            if totalSum >= target:
                minNumElements = min(minNumElements, right - left + 1)
                totalSum -= nums[left]
                left += 1
            else:
                right += 1

                if right < len(nums):
                    totalSum += nums[right]

        if minNumElements == float('inf'):
            return 0
        
        return minNumElements

File: Data Structures & Algorithms/jump-game-ii/submission-1.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int len = nums.size();
        nums[len - 1] = 0;

        for (int i = len - 2; i >= 0; i--) {
            int minJumps = 1e9;
            int itLimit = min(i + nums[i] + 1, len);

            for (int j = i + 1; j < itLimit; j++) {
                minJumps = min(minJumps, nums[j]);
            }

            nums[i] = minJumps + 1;
        }

        return nums[0];
    }
};
